### DISCUSSION TITLE
Integrating Emscripten with PouchDB

### DISCUSSION THREAD

**Discussion by researchproposal552** [pouchdb/pouchdb#8472] [Q&A]
I am using CPP as programming language. I would like to know can Emscripten be integrate with PouchDB? If yes, how is it possible?

**Comment by SourceR85** [comment]
Did you mean, that you would like to interact with PouchDB in a WASM module?

Short answer: you shouldn't

<details><summary>Details</summary>
<p>
You have to bind the export table of your wasm instance to the methods of a pouch instance, if you want to call em directly (it's a much better idea to write yourself some helpers in JS and bind these instead).

You also need to transfer (pack in js -> unpack in c++) every value crossing these interface. This can be very slow!
</p>
</details>

I'll close this, since your GH history is empty and I didn't expect any other reaction to this discussion.

If you're still interested in PouchDB + WASM feel free to open another thread with more details (best case: you including a link to your demo project so others can play with you on that idea) 😉
